src/Cache/Adapter/Mcrouter.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+namespace Utopia\Cache\Adapter;
+
+use Utopia\Cache\Adapter;
+use Mcrouter\Mcrouter;
+
+class Mcrouter implements Adapter
+{
+    /**
+     * @var Mcrouter
+     */
+    protected Mcrouter $mcrouter;
+
+    /**
+     * Mcrouter constructor.
+     *
+     * @param  Mcrouter  $mcrouter
+     */
+    public function __construct(Mcrouter $mcrouter)
+    {
+        $this->mcrouter = $mcrouter;
+    }
+
+    /**
+     * @param  string  $key
+     * @param  int  $ttl time in seconds
+     * @return mixed
+     */
+    public function load(string $key, int $ttl): mixed
+    {
+        $result = $this->mcrouter->get($key);
+
+        if ($result->getResult() !== null) {
+            $cache = json_decode($result->getResult(), true);
+
+            if ($cache['time'] + $ttl > time()) { // Cache is valid
+                return $cache['data'];
+            }
+        }
```

Comment:
It might be good to exit early to reduce nesting just like the redis' load method:

https://github.com/utopia-php/cache/blob/5b3691b86338dc63835d325003d8ebcd0043c492/src/Cache/Adapter/Redis.php#L33-L36